packages/terra-section-header/src/SectionHeader.jsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import Arrange from 'terra-arrange';
+import styles from './SectionHeader.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Text to be displayed on the SectionHeader.
+   */
+  title: PropTypes.node.isRequired,
+  /**
+   * Callback function triggered when the accordion icon is clicked.
+   */
+  onClick: PropTypes.func,
+  /**
+   * Whether the accordion icon should be displayed in its 'open' or 'closed' position.
+   */
+  isOpen: PropTypes.bool,
+};
+
+const defaultProps = {
+  onClick: undefined,
+  isOpen: false,
+};
+
+const SectionHeader = ({
+  title,
+  onClick,
+  isOpen,
+  ...customProps
+}) => {
+  if ((process.env.NODE_ENV !== 'production') && (!onClick && isOpen)) {
```

Comment:
Can we add this message to the isOpen prop description and then remove this warning?

Reply:
https://github.com/cerner/terra-core/pull/1307/files/48ab9ff3edd3f5c7009714327ec86c4e254ded09#r171583836